src/App/App.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from "react";
+import "./App.css";
+import { getBookList } from "../apiCalls";
+import BookList from "../BookList/BookList";
+import { Route, NavLink } from "react-router-dom";
+import { SearchBar } from "../SearchBar/SearchBar";
+import FullBook from "../FullBook/FullBook";
+import { WelcomePage } from "../WelcomePage/WelcomePage";
+import { SearchResults } from "../SearchResults/SearchResults";
+
+const App = () => {
+  const [bookList, setBookList] = useState({});
+  const [userInput, setUserInput] = useState("");
+  const [userSearchResults, setUserSearchResults] = useState([]);
+
+  useEffect(() => {
+    getBookList().then((response) => {
+      const acceptableFormats = response.results.filter((item) => {
+        if (item.formats["text/html"] !== undefined) {
```

Comment:
`!== undefined` is truthy. Could you use the bang operator instead for this conditional?

Reply:
I thought there was some rule about using bang operators with object's properties that make it not work or something?  Need to do further research.